src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -64,6 +64,11 @@ interface DiscoverStateContainerParams {
    * core ui settings service
    */
   services: DiscoverServices;
+  /*
+   * mode in which discover is running
+   *
+   * */
+  mode?: DiscoverDisplayMode;
```

Comment:
This makes sense, embedded | standalone. What do you think about a 2nd prop, an appId to redirect if mode === 'embedded'

Reply:
yes, for determining path, if discover is being used at a different path in both security and observability serverless modes.

Reply:
guess that might not work with locators and the like, as those run during discover setup. Probably makes more sense to do something at a config level then? maybe a discover.location setting that we set only in serverless.obit.yml/serverless.security.yml?

Reply:
Does it make sense to add parameter in `discover.locator` called, appId where locator will develop the URL based on the appID given?

Reply:
> Observability currently doesn't use Discover at a different path or embedded in another app. They're using the `profile` functionality in the customization framework to customize Discover as a top level app, so their use case around URLs and paths is a bit different (and already supported).

In light of today's Discover Working Stream, scratch this -- looks like Observability will be embedding the Discover container now (I just learned of this now). With that said, they seem to be planning to include the main Discover app in their project, so their use case around URLs and redirecting is still different since existing links will presumably continue to take users to the main Discover app.

Reply:
@davismcphee thanks for the info. What do you think if client can pass callback to locator to provide a client specific location. This will prevent any other plugin's logic in Discover?

cc:  @kqualters-elastic